trie/proof.libevm.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package trie
+
+import (
+	"errors"
+)
+
+var errKeysHaveDifferentLengths = errors.New("keys have different lengths")
```

Comment:
nit may I suggest to NOT have a sentinel error, and use a simple inline string, given this is not used in production code, to reduce the Go API (and files added)? The test can just check the error string matches a certain regex string such as `^keys have different lengths`... optional though

Reply:
I do not think that we should copy paste the string between the test and the code.

I also do not think that we should have a const string rather than an error.

What is the benefit to using textual errors?

Reply:
Going to mark this as resolved as it was previously approved by the rest of the team